tree/ntuple/v7/src/RDaos.cxx

```diff
@@ -64,32 +64,39 @@ std::string ROOT::Experimental::Detail::RDaosObject::ObjClassId::ToString() cons
 }
 
 ROOT::Experimental::Detail::RDaosObject::FetchUpdateArgs::FetchUpdateArgs(FetchUpdateArgs &&fua)
-   : fDkey(fua.fDkey), fAkey(fua.fAkey), fIods{fua.fIods[0]}, fSgls{fua.fSgls[0]}, fIovs(std::move(fua.fIovs)),
+   : fDkey(fua.fDkey), fAkeys(fua.fAkeys), fIovs(fua.fIovs), fIods(std::move(fua.fIods)), fSgls(std::move(fua.fSgls)),
      fEvent(std::move(fua.fEvent))
 {
    d_iov_set(&fDistributionKey, &fDkey, sizeof(fDkey));
-   d_iov_set(&fIods[0].iod_name, &fAkey, sizeof(fAkey));
 }
 
-ROOT::Experimental::Detail::RDaosObject::FetchUpdateArgs::FetchUpdateArgs(DistributionKey_t &d, AttributeKey_t &a,
-                                                                          std::vector<d_iov_t> &v, bool is_async)
-   : fDkey(d), fAkey(a), fIovs(v)
+ROOT::Experimental::Detail::RDaosObject::FetchUpdateArgs::FetchUpdateArgs(const DistributionKey_t &d,
+                                                                          std::span<AttributeKey_t const> as,
+                                                                          std::span<d_iov_t const> vs, bool is_async)
+   : fDkey(d), fAkeys(as), fIovs(vs)
 {
    if (is_async)
       fEvent.emplace();
 
+   fSgls.reserve(fAkeys.size());
+   fIods.reserve(fAkeys.size());
    d_iov_set(&fDistributionKey, &fDkey, sizeof(fDkey));
 
-   d_iov_set(&fIods[0].iod_name, &fAkey, sizeof(fAkey));
-   fIods[0].iod_nr = 1;
-   fIods[0].iod_size = std::accumulate(v.begin(), v.end(), 0,
-                                       [](daos_size_t _a, d_iov_t _b) { return _a + _b.iov_len; });
-   fIods[0].iod_recxs = nullptr;
-   fIods[0].iod_type = DAOS_IOD_SINGLE;
-
-   fSgls[0].sg_nr_out = 0;
-   fSgls[0].sg_nr = fIovs.size();
-   fSgls[0].sg_iovs = fIovs.data();
+   for (unsigned i = 0; i < fAkeys.size(); ++i) {
+      daos_iod_t iod;
+      iod.iod_nr = 1;
+      iod.iod_size = fIovs[i].iov_len;
+      iod.iod_recxs = nullptr;
+      iod.iod_type = DAOS_IOD_SINGLE;
+      d_iov_set(&iod.iod_name, const_cast<AttributeKey_t *>(&(fAkeys[i])), sizeof(fAkeys[i]));
+      fIods.push_back(iod);
+
+      d_sg_list_t sgl;
+      sgl.sg_nr_out = 0;
+      sgl.sg_nr = 1;
+      sgl.sg_iovs = const_cast<d_iov_t *>(&fIovs[i]);
```

Comment:
This `const_cast` shouldn't be needed after removing the `const` from the `std::span<const d_iov_t>`, as said above.